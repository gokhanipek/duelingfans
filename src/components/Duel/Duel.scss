.duel__page--wrapper {
    display: flex;
    width: 100%;
    height: 100vh;
    padding-top: 150px;
    flex-wrap: wrap;
    align-items: center;
    flex-direction: column;
    @media screen and (max-width: 992px) {
        flex-wrap: nowrap;
        padding-top: 10px;
    }
    .duel__basketball--wrapper {
        width: 500px;
        @media screen and (max-width: 992px) {
            width: 70%;
        }
        .duel__basketball {
            width: 100%;
            height: 70px;
            background-image: url('./../../assets/basketball.png');
            background-repeat: no-repeat;
            background-position: center;
            background-position-x: 0;
            background-size: 200%;
        }
    }
    .duel__player--wrapper {
        flex-direction: row;
        display: flex;
    }
    .duel__score {
        font-size: 2em;
        font-family: 'Shentox';
        color: #1f375b;
        @media screen and (max-width: 992px) {
            margin: 0;
        }
        span {
            color: #fff;
        }
    }
    .duel__page--button {
        margin: 0;
        padding: 1.5em 0;
        width: 45%;
        margin-top: 20px;
        max-width: 300px;
        border-radius: 6px;
        border: 2px solid #1493ff;
        background: #1f375b;
        font-family: "Shentox";
        font-size: .75rem;
        line-height: 1;
        text-align: center;
        text-transform: uppercase;
        box-shadow: 0 4px 6px rgba(0, 0, 0, 0.4);
        -webkit-transition: all .2s ease;
        text-decoration: none;
        color: #fff;
        transition: all .2s ease;
        position: relative;
        z-index: 1;
        cursor: pointer;
    }
}