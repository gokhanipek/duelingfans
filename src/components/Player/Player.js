import React from 'react';
import './Player.scss';

const Player = ({name, image, fppg, onClickHandler, showFppg}) => {
    return (
        <div className="player__page--wrapper" id="player-card" onClick={onClickHandler}>
            <div className="player__card">
                {showFppg && <p className="player__fppg">{ fppg || 0 }</p>}
                <img className="player__card--image" src={image} alt="player" />
            </div>
            <p className="player__name">{name}</p>
        </div>
    )
}

export default Player;
