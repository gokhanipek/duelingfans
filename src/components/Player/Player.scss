.player__page--wrapper {
    display: flex;
    flex-direction: column;
    cursor: pointer;
    transition: transform 0.4s;
    @media screen and (max-width: 992px) {
        margin-bottom: 10px;
    }
    .player__card {
        position: relative;
        width: 300px;
        height: 300px;
        border: 1px solid #1493ff;
        overflow: hidden;
        background-image: url('./../../assets/bg.png');
        background-size: cover;
        background-repeat: no-repeat;
        box-shadow: 0 0 6px 0px #1f375b;
        margin: 0px 10px;
        @media screen and (max-width: 992px) {
            width: 150px;
            height: 150px;
            margin: 0px 5px;
        }
        .player__card--image {
            width: 100%;
        }
        .player__fppg {
            position: absolute;
            left: 0;
            top: 0;
            width: 100%;
            margin: 0;
            padding: 5px;
            color: #fff;
            background-color: #1f375b;
        }
    }
    .player__name {
        font-family: 'Shentox';
        font-size: 1em;
        color: #fff;
        margin: 0px 10px;
        background-color: #1f375b;
        padding: 10px;
        @media screen and (max-width: 992px) {
            margin: 0px 5px;
        }
    }
    &:hover {
        transform: scale(1.02);
    }
}