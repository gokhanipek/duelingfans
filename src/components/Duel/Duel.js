import React, {useEffect} from 'react';
import Player from './../Player/Player';
import { withRouter } from 'react-router-dom';
import { connect } from "react-redux";
import { getPlayerListRequest, incrementScore, incrementTurn, hideFppg } from './../../redux/actions';
import './Duel.scss';

const Duel = ({players, getPlayerListRequest, hideFppg, incrementScore, incrementTurn, score, turn, showFppg, history}) => {

    useEffect(() => {
      getPlayerListRequest();
    }, []);

    const style = {
      backgroundPositionX: `${turn !== 0 ? turn * 10: 0}%`
    };

    const onClickHandler = fppg => {
      if(showFppg) return;
      const highestNumber = players.length > 0 && Math.max(...players.map(player => player.fppg))
      fppg >= highestNumber ? incrementScore() : incrementTurn();
    }

    const nextRound = () => {
      if(turn > 9) history.push(`/result&${score > 9 ? 'success' : 'fail'}`)
      hideFppg();
      getPlayerListRequest();
    }

    return (
      <div className="duel__page--wrapper">
          <div className="duel__basketball--wrapper">
            <div className="duel__basketball" style={style}/>
          </div>
          <p className="duel__score">Your score: <span>{score}</span></p>
          <div className="duel__player--wrapper">
            {players.length > 0 && players.map(player =>
              <Player
                name={player.first_name + ' ' + player.last_name}
                image={player.images.default.url}
                fppg={player.fppg}
                showFppg={showFppg}
                onClickHandler={() => onClickHandler(player.fppg)}
              />)}
          </div>
          <div className="duel__page--button" onClick={() => nextRound()}>Next</div>
      </div>
    );
}

const mapDispatchToProps = dispatch =>({
  getPlayerListRequest: () => dispatch(getPlayerListRequest()),
  incrementScore: () => dispatch(incrementScore()),
  incrementTurn: () => dispatch(incrementTurn()),
  hideFppg: () => dispatch(hideFppg()),
})
const mapStateToProps = ({data}) => ({ 
  players: data.playerList,
  turn: data.turn,
  score: data.score,
  showFppg: data.showFppg
});
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Duel));