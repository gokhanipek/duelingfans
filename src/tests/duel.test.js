import React from 'react';
import { shallow  } from 'enzyme';
import Duel from './../components/Duel/Duel';
import Player from './../components/Player/Player';

describe('Duel component', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Duel debug />);
  
    expect(component).toMatchSnapshot();
  });

  it('does not render <Player /> component if no player available', () => {
    const wrapper = shallow(<Duel />);
    expect(wrapper.find(Player)).toHaveLength(0);
  });
});

  