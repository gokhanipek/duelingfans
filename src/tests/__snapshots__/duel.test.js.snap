// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Duel component should render correctly in "debug" mode 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
